### ISSUE TITLE
Check lengths of `RegularInTime` arguments?

### ISSUE THREAD

**Issue by simsurace** [JuliaGaussianProcesses/TemporalGPs.jl#119]
I wonder whether the `RegularInTime` constructor should error if the lengths of its arguments don't match:

https://github.com/JuliaGaussianProcesses/TemporalGPs.jl/blob/3c56bebabeaf9a5cf602c26c169b448c028ce7d9/src/space_time/regular_in_time.jl#L8-L13

**Comment by willtebbutt** [comment]
Good catch -- that would be ideal.

This kind of check should be shoved inside another function that is marked `@non_differentiable` so that it doesn't hit AD performance.

**Comment by simsurace** [comment]
Interesting... is the constructor function ever AD'd through? I would have thought that once the object has been constructed, the AD wouldn't be able to see the function that was called during constuction.

**Comment by willtebbutt** [comment]
That's true -- I was envisaging the situation in which one happens to construct the `RegularInTime` inside the function that you're differentiating.
